plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/VerifierErrorInterpreter.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver
+
+import org.jetbrains.kotlin.formver.viper.errors.*
+
+/**
+ * TODO: currently the error messages are returned as plain string, not using the `PluginErrors` object.
+ *       We have to investigate if it is better to use `VIPER_VERIFICATION_ERROR` only,
+ *       or creating an error for each possible type.
+ */
+
+object VerifierErrorInterpreter {
+
+    private fun ConsistencyError.toHumanReadableMessage(debug: Boolean): String = when (debug) {
+        true -> this.msg
+        false -> "An internal error has occurred while verifying the function. Please report the issue."
+    }
+
+    private fun VerificationError.toHumanReadableMessage(debug: Boolean): String {
+        val context = lookupForEmbeddedInfo<ContextErrorInfo>(this, ContextErrorInfo.Unknown)
+        return when {
+            this is PostconditionViolated && context is ContextErrorInfo.CondBoolEffect -> "the following boolean condition might not hold."
+            this is PostconditionViolated && context is ContextErrorInfo.CondNullEffect -> "the following nullability condition might not hold."
+            this is PostconditionViolated && context is ContextErrorInfo.CondNotNullEffect -> "the following non-nullability condition might not hold."
```

Comment:
It would be nice to have a more concrete message here. If a user says returns(true), then the warning "function may return false" is more useful than "the condition might not hold".

Reply:
I agree! I should store some extra values on the `ContextErrInfo`

Reply:
@jesyspa if we have the following function:

```kotlin
@OptIn(ExperimentalContracts::class)
fun <!VIPER_TEXT!>non_nullable_returns_null<!>(x: Int): Int {
    contract {
        <!VIPER_VERIFICATION_ERROR!>returns(null)<!>
    }
    return x
}
```

Should we get an error message like: `function may return a non-nullable value.`?

Reply:
`non-nullable` is a property of types, not values; this would be something like "function may return a non-null value" or "a value other than null".